pi-manage

```diff
@@ -1288,75 +1288,82 @@ def import_tokens(file, tokenrealm=None):
+                            '{0!s}'.format(', '.join(['all'] + list(EXPORT_FUNCTIONS.keys()))))
 def exporter(output, fmt, types):
     """export configuration"""
-    print(output, file=sys.stderr)
-    print(fmt, file=sys.stderr)
-    print(types, file=sys.stderr)
     exp_types = EXPORT_FUNCTIONS.keys() if 'all' in types else types
 
+    out = {}
     for typ in exp_types:
-        res = {typ: EXPORT_FUNCTIONS[typ]()}
-        res = fmt_dict.get(fmt)(res) + '\n'
+        out.update({typ: EXPORT_FUNCTIONS[typ]()})
+
+    if out:
+        res = exp_fmt_dict.get(fmt.lower())(out) + '\n'
         output.write(res)
 
 
-inp_fmt_dict = {'python': ast.literal_eval,
+imp_fmt_dict = {'python': ast.literal_eval,
                 'json': json.loads,
                 'yaml': yaml.safe_load}
 
 
 @config_manager.option('-i', '--input', type=argparse.FileType('r'),
                        default=sys.stdin, dest='infile',
-                       help='The filename to import the data from,'
-                            ' default is <stdin>')
+                       help='The filename to import the data from. Try to read '
+                            'from <stdin> if this argument is not given.')
 @config_manager.option('-t', '--types', nargs='*', default=['all'],
+                       choices=['all'] + list(IMPORT_FUNCTIONS.keys()),
                        help='The types of configuration to import. By default import all '
                             'available data if a corresponding importer type exists. '
-                            'Currently registered importer types '
-                            'are: {0!s}'.format(', '.join(IMPORT_FUNCTIONS.keys())))
+                            'Currently registered importer types are: '
+                            '{0!s}'.format(', '.join(['all'] + list(IMPORT_FUNCTIONS.keys()))))
 def importer(infile, types):
     """import configuration"""
-    d = None
+    data = None
     imp_types = IMPORT_FUNCTIONS.keys() if 'all' in types else types
 
     content = infile.read()
 
-    for tp in inp_fmt_dict:
+    for fmt in imp_fmt_dict:
         try:
-            d = inp_fmt_dict[tp](content)
+            data = imp_fmt_dict[fmt](content)
             break
-        except (SyntaxError, json.decoder.JSONDecodeError, yaml.error.YAMLError) as e:
+        except (SyntaxError, json.decoder.JSONDecodeError, yaml.error.YAMLError) as _e:
             continue
-    if not d:
-        print('Could not read input format! Accepting JSON, YAML or native Python.')
+    if not data:
+        print('Could not read input format! '
+              'Accepting: {0!s}.'.format(', '.join(imp_fmt_dict.keys())),
+              file=sys.stderr)
         sys.exit(1)
 
-    for typ, content in d.items():
+    # we need to go through the importer functions based on priority
+    for typ, value in sorted(IMPORT_FUNCTIONS.items(), key=lambda x: x[1]['prio']):
         if typ in imp_types or 'all' in imp_types:
```

Comment:
I think this only needs to be:

~~~~python
if typ in imp_types:
~~~~
since in line `imp_types` was already set to the `.keys()` if 'all' as contained.
Thus, 'all' can not be contained in `imp_types` at all.

Reply:
Done in f6f28b91